tests/src/cache_tests.cpp

```diff
@@ -377,3 +383,81 @@ TEST(CacheTest, SingleThreadSwapDoesntThrow)
     using std::swap;
     swap(cache_a, cache_b);
 }
+
+struct CompositeKey {
+    std::string first_part;
+    std::string second_part;
+
+    template<typename H> friend H AbslHashValue(H h, const CompositeKey& s)
+    {
+        return H::combine(std::move(h), s.first_part, s.second_part);
+    }
+
+    bool operator==(const CompositeKey& other) const
+    {
+        return first_part == other.first_part && second_part == other.second_part;
+    }
+};
+
+struct CompositeKeyView {
+    std::string_view first_part;
+    std::string_view second_part;
+
+    bool operator==(const CompositeKeyView& other) const
+    {
+        return first_part == other.first_part && second_part == other.second_part;
+    }
+
+    bool operator==(const CompositeKey& other) const
+    {
+        return first_part == other.first_part && second_part == other.second_part;
+    }
+
+    template<typename H> friend H AbslHashValue(H h, const CompositeKeyView& s)
+    {
+        return H::combine(std::move(h), s.first_part, s.second_part);
+    }
+};
+
+TEST(CacheTest, FindHeterogeneousLookup)
+{
+    // Can hash either a `CompositeKey` with `absl::Hash` or a `CompositeKeyView` with absl::Hash.
+    using Hasher = MultiHash<CompositeKey, absl::Hash<CompositeKey>, CompositeKeyView, absl::Hash<CompositeKeyView>>;
+
+    using TestCache = presets::memory::TinyLFUCache<CompositeKey, Point3D, measurement::SizeOf<Point3D>, measurement::SizeOf<CompositeKey>, Hasher>;
+
+    TestCache cache{100 * sizeof(Point3D)};
+
+    CompositeKey     key{"asdf", "hjkl"};
+    CompositeKeyView key_view{"asdf", "hjkl"};
+
+    Point3D value{1, 1, 1};
+
+    cache.find(key_view);
+    cache.insert(key, value);
```

Comment:
Why do you call `find` before inserting? Do you want to show that it doesn't throw? Maybe you should add a comment to clarify your intent. :thinking: Or you could verify that the `find` returned "not found".